### PR TITLE
Fix: "Test failing in `beforeEach` is not reported as failed"

### PR DESCRIPTION
**Pull request by AnnikaNissen**
See issue https://github.com/sullenor/wdio-teamcity-reporter/issues/21 

This PR makes the reporter:
* report failed hooks as failed tests
* report tests pending on suite end as failed

This leads to, that errors in fx `beforeEach` or `afterEach` hooks are properly visible in the TeamCity UI.

### CODE REVIEW

**Comment by AnnikaNissen** [comment]
@sullenor Can you help me getting the test up & running on Travis?
```
[0-1]  Error:  Failed to create session.
Unable to connect to "http://localhost:4444/wd/hub", make sure browser driver is running on that address.
If you use services like chromedriver see initialiseServices logs above or in wdio.log file as the service might had problems to start the driver.
```

File: index.js [hunk 1]
```diff
@@ -125,6 +149,15 @@ class WdioTeamcityReporter extends WdioReporter {
    * @param {SuiteStats} suiteStats
    */
   onSuiteEnd (suiteStats) {
+    const pendingTests = Object.values(suiteStats.tests).filter(test => test.state === 'pending')
+    pendingTests.forEach(testStat => {
+      testStat.error = testStat.error || {
+        message: '',
```
Comment (sullenor) [inline index.js:155]: Can we show a default message in this case?

Comment (sullenor) [inline index.js:155]: This one reports pending tests as failed. How do you think, may be we can put a default message here instead?

**Comment by sullenor** [comment]
I have a small concern related to an empty message, but overall looks good to me. 
I also checked travis issue: updated dependencies and updated chrome version https://github.com/sullenor/wdio-teamcity-reporter/pull/23. Should be good for now.

File: index.js (hunk 1 above)
Comment (AnnikaNissen) [reply index.js]: Good point. I added a default message.

**Comment by sullenor** [comment]
Thanks @AnnikaNissen for your outstanding work :) Your patch should be available with version 2.3.0.

**Comment by AnnikaNissen** [comment]
You are welcome, @sullenor ! Thanks for your help with the build setup and for the quick merge :-)

**Pull request merged**